# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding and white-sounding names
print ("White_sounding name calls:", np.sum(data[data.race=='w'].call))
print ("Black-sounding name calls:", np.sum(data[data.race=='b'].call))

White_sounding name calls: 235.0
Black-sounding name calls: 157.0


In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

### 1. What test is appropriate for this problem? Does CLT apply?

The outcome of the observations are either 0 or 1 (Bernouli random variable) and the whole problem is binomial distribution which is the sum of independent and identically distributed Bernouli random variables. Since there are two sample sets (black and white sounding names) and there are enough observations (and no population standard deviation) we can use two-sample t_test on data set. 

The central limit theorem (CLT) is a statistical theory that states that given a sufficiently large sample size from a population with a finite level of variance, the mean of all samples from the same population will be approximately equal to the mean of the population. Furthermore, all of the samples will follow an approximate normal distribution pattern, with all variances being approximately equal to the variance of the population divided by each sample's size.
As a general rule, sample sizes equal to or greater than 30 are considered sufficient for the central limit theorem to hold, meaning the distribution of the sample means is fairly normally distributed.





### 2. What are the null and alternate hypotheses?

Our statistic is the rate of successful draw (X=1). Under the null hypothesis, we consider that the success rate (call on resume) of black-sounding names is similar to success rate of white-sounding names. 
Alternative hypothesis is the success rate of black-sounding names is different than white-sounding names. 

In [6]:
blk_df = data[data['race'] =='b']
wh_df = data[data['race'] == 'w']

tot_blk = len(blk_df)
tot_wh = len(wh_df)

print ("Total black resumes and white resumes are:", tot_blk, "and", tot_wh)

num_blk_call = np.sum(blk_df['call']==1)
num_wh_call = np.sum(wh_df['call']==1)
num_diff = num_wh_call - num_blk_call
print ("Number of calls on black and white are:" , num_blk_call, "and", num_wh_call)

rate_blk_call = (num_blk_call)/(tot_blk)
rate_wh_call = (num_wh_call) / (tot_wh)

print ("Black call rate is: %F"%rate_blk_call)
print ("White call rate is: %F" %rate_wh_call)

rate_diff_wh_blk = rate_wh_call - rate_blk_call
print ("Difference in rate of white and black resume calls is:%F" %rate_diff_wh_blk)

Total black resumes and white resumes are: 2435 and 2435
Number of calls on black and white are: 157 and 235
Black call rate is: 0.064476
White call rate is: 0.096509
Difference in rate of white and black resume calls is:0.032033


### 3. Compute margin of error, confidence interval, and p-value.

Margin of error = Critical value x Standard error of the sample. 

Critical Value can be either z-score or t-score. In this problem I consider 95% confidence interval and its z score is 1.96. 

In [7]:
var_wh = (rate_wh_call)*(1-rate_wh_call) / tot_wh 
var_blk = (rate_blk_call)*(1-rate_blk_call) / tot_blk

print (var_wh , var_blk)

# Sample variance 

var_wh_blk = var_wh + var_blk 
std_wh_blk = np.sqrt(var_wh_blk)

# Margin of error 
marg_error = 1.96 * std_wh_blk
print ("Margin of error:", marg_error)

# Confidence interval
conf_interval_beg =  rate_diff_wh_blk - marg_error 
conf_interval_end =  rate_diff_wh_blk + marg_error

print ("Confidence interval is [%F , %F]" %(conf_interval_beg ,conf_interval_end))

# P-value 

#sample_concat = np.concatenate((blk_df['call'],wh_df['call']))
bs_replicates = np.empty(10000)

for i in range(10000):
    perm_all = np.random.permutation(data['call'])
    wh_perm = perm_all[:2435]
    wh_rep_rate = np.sum(wh_perm)/(len(wh_df))
    blk_perm = perm_all[2435:]
    blk_rep_rate = np.sum(blk_perm)/(len(blk_df))
    bs_replicates[i] = wh_rep_rate - blk_rep_rate
    

print (np.sum(blk_df['call']))
#print (bs_replicates) 
print ("P-Value is : %F" %((np.sum(bs_replicates>rate_diff_wh_blk))/len(bs_replicates)))

3.5809119833e-05 2.47717378565e-05
Margin of error: 0.0152554063499
Confidence interval is [0.016777 , 0.047288]
157.0
P-Value is : 0.000000


P_value is very very small, so it shows that our null hypothesis is rejected and there is difference between black and white nsounding names' resumes calls. 

### 4. Write a story describing the statistical significance in the context or the original problem.

Considering 95% confidence level, the confidence intervall of different between the rate of call on white sounding names and black sounding names is the range of 1.68% - 4.7%. 
In other words, if we keep everything fixed and just use two different names (one white sounding and the other black sounding) the chance that we get the call for white version is 1.68% - 4.7% more than getting called on black version (with 95% confidence level). This difference is statistically significant (calculations shown above). 
To sum it up, if someone just change the resume name from black sounding to white sounding his/her chance of being called would increase from 6.4% to 9.6% on average and with 95% confidence level the increased chance would be within the range of 1.68% to 4.7% (which comparing to 6.4% is a huge increase). 

### 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

Not exacatly. Actually this piece just focused on the name (and its race sounding feature) of the resumes. It does not get into the other features of a resume. Just by this analysis we can not conclude anything about the most important factor in callback success. If we want to expand our analysis on resumes and their chance of getting called we need to include all the possibe factors that may affect the success rate. These factors can be age / level of education / gender / area of living (country, county, city,...) / work experience / field of work / and so on. 